docs/internal/DistributedArchitectureGuide.md

```diff
@@ -12,6 +12,214 @@ A guide to the general Elasticsearch components can be found [here](https://gith
 
 # Networking
 
+Every elasticsearch node maintains various networking clients and servers,
+protocols, and synchronous/asynchronous handling.
+
+## HTTP Transport
+
+Every node has an HTTP Transport Server (simply HTTP Transport) and it is a
+single entry point for all external clients (except cross-cluster). Management,
+ingestion, search, and everything else passes through HTTP server.
+
+Elasticsearch works over HTTP 1.1 and supports HTTP features such as: TLS,
+chunked transfer encoding, content compression, pipelining. Although we try to
+be HTTP spec compliant, Elastic is not a webserver. We support GET requests with
+payload (some old proxies might drop content), requests cannot be cached by
```

Comment:
Maybe mention that in these cases we also support the same API with a different verb, normally POST, for clients who cannot send a GET-with-body.